File: life-admin/src/lib/api/gamification.ts
import { apiClient } from './client';

export interface GamificationState {
  id: string;
  user_id: string;
  xp_total: number;
  level: number;
  level_title: string;
  next_level_xp: number | null;
  xp_to_next_level: number | null;
  streak_current: number;
  streak_longest: number;
  streak_last_date: string | null;
  streak_freezes: number;
  badges: unknown[];
  daily_missions: unknown[];
}

export async function getGamification(): Promise<GamificationState> {
  const { data } = await apiClient.get<GamificationState>('/gamification/me');
  return data;
}

File: alia-backend/app/models/gamification.py
from datetime import date
from typing import Any
from uuid import UUID

from pydantic import BaseModel


class GamificationResponse(BaseModel):
    """Gamification state returned to the client."""

    id: UUID
    user_id: UUID
    xp_total: int
    level: int
    level_title: str
    next_level_xp: int | None = None
    xp_to_next_level: int | None = None
    streak_current: int
    streak_longest: int
    streak_last_date: date | None = None
    streak_freezes: int
    badges: list[Any]
    daily_missions: list[Any]
